### PR TITLE
[CDAP-7974] Handle unknown entities in CDAP UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds new Page404 component to be used everywhere to represent unknown entities.
- Adds behavior to handle unknown namespace, apps, datasets, streams 

JIRA: https://issues.cask.co/browse/CDAP-7974
Bamboo build: http://builds.cask.co/browse/CDAP-DRC5572/latest

### CODE REVIEW

File: cdap-ui/app/cdap/components/EntityListView/index.js [hunk 1]
```diff
@@ -536,6 +567,38 @@ class EntityListView extends Component {
...
+            entityName={this.props.params.namespace}
+          >
+            <div className="namespace-not-found text-xs-center">
+              <h4>
+                <strong>
+                  {T.translate('features.EntityListView.NamespaceNotFound.createMessage')}
+                  <span
+                    className="open-namespace-wizard-link"
+                    onClick={() => {
+                      console.log(this);
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/EntityListView/index.js]: remove?

File: cdap-ui/server/config/ui-settings.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
   "standalone.website.sdk.download": "true",
-  "ui.debug.enabled": "true"
+  "ui.debug.enabled": "false"
```
Comment (tonybach) [inline cdap-ui/server/config/ui-settings.json:3]: not sure what this does, but did you intend to push this change?

File: cdap-ui/app/cdap/components/StreamDetailedView/index.js [hunk 3]
```diff
@@ -77,35 +79,45 @@ export default class StreamDetailedView extends Component {
...
+              entityDetail
+            }, () => {
+              setTimeout(() => {
+                this.setState({
+                  loading: false
+                });
+              }, 1000);
+            });
+          },
+          err => {
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/StreamDetailedView/index.js]: should use the same convention with `res` on line 83? Appeared in a few other places as well.

**Comment by tonybach** [comment]
I think the pages look great! Just some minor comments about the code.

**Comment by elfenheart** [comment]
screen shot 2017-02-07 at 6 11 03 pm

Should we just hide the active Namespace section if it's not found?

screen shot 2017-02-07 at 6 11 30 pm

Also I try to navigate to a non-existing namespace and non-existing stream. It should just go to the Namespace level 404 right?

**Comment by tonybach** [comment]
Just merged my search/sort/filter PR, so it created some merge conflicts for you (oops...). But now you can see how it looks like with both the search/sort bar and your new nav 😃

File: cdap-ui/server/config/ui-settings.json (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/server/config/ui-settings.json:3]: @tonybach yes. This allows UI to extensively log in browser console. Its not necessary for production. Its a dev only setting for us.

**Comment by ajainarayanan** [comment]
@elfenheart I did not disable the dropdown as the user could open the namespace dropdown and choose another namespace.

And reg #2 you are right. This is what too me so long to create a PR. Right now I am running out of ideas to detect hierarchical check for entities (namespace, app, dataset, stream) while navigating to a state. Will try to play around tonight

**Comment by elfenheart** [comment]
@ajainarayanan not disabling the dropdown, but just hiding the active namespace card, since it looks like that `--` namespace is selected.

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**